server.js

```diff
@@ -3414,27 +3414,55 @@ cache(function(data, match, sendBadge, request) {
   var repo = match[2];  // eg, subtitleedit
   var version = match[3];  // eg, 3.4.7
   var format = match[4];
-  var apiUrl = githubApiUrl + '/repos/' + user + '/' + repo + '/compare/' + version + '...master';
   var badgeData = getBadgeData('commits since ' + version, data);
-  if (badgeData.template === 'social') {
-    badgeData.logo = getLogo('github', data);
-  }
-  githubAuth.request(request, apiUrl, {}, function(err, res, buffer) {
-    if (err != null) {
-      badgeData.text[1] = 'inaccessible';
-      sendBadge(format, badgeData);
-      return;
-    }
-    try {
-      var data = JSON.parse(buffer);
-      badgeData.text[1] = metric(data.ahead_by);
-      badgeData.colorscheme = 'blue';
-      sendBadge(format, badgeData);
-    } catch(e) {
-      badgeData.text[1] = 'none';
-      sendBadge(format, badgeData);
+
+  function setCommitsSinceBadge(user, repo, version) {
```

Comment:
Could you move this to `lib/github-badge-helpers.js` like I've done with the other badge-specific functions in #1109?

Reply:
I've checked the #1109 but `setCommitsSinceBadge()` is not helper function. `setCommitsSinceBadge()` is responsible for actual output. As I've maintained before, we've to call 2 send two times `/GET` request, so that I've separated the existing  code to a function & it after getting that latest version from server, I've called the function. 

please check the `if-else` block.

I think, it is better to separate whole code to `lib/github-badge-provider`.

What you think? . 😄

Reply:
Yup, makes sense to me. That's consistent with @Daniel15's input in https://github.com/badges/shields/pull/1109#discussion_r142043932. I ended up naming them e.g. `npm-provider.js`, so `github-provider.js` would be consistent with that.